server/src/main/java/org/elasticsearch/cluster/metadata/Metadata.java

```diff
@@ -1305,16 +1305,10 @@ public Map<String, ComposableIndexTemplate> templatesV2() {
             .orElse(Collections.emptyMap());
     }
 
+    // TODO: remove this method:
```

Comment:
This can be removed once the code that invokes index settings providers use `retrieveIndexModeFromTemplate(...)` instead.